falcon/util/uri.py

```diff
@@ -292,9 +292,9 @@ def parse_query_string(query_string, keep_blank_qs_values=False):
             # a list, or append the next value to the list.
             old_value = params[k]
             if isinstance(old_value, list):
-                old_value.append(v)
+                old_value.append(decode(v))
             else:
-                params[k] = [old_value, v]
+                params[k] = [old_value, decode(v)]
 
         else:
             if ',' in v:
```

Comment:
Ah, good point. Makes sense. Thanks for the clarification! Would you mind adding a `# NOTE(steffgrez): blah blah` comment to the code about this?